### PR TITLE
modernize packaging for setuptools>30.3

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
ref #3266 

this starts a build tooling cleanup,

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -104,14 +33,12 @@ def main():
             "Floris Bruynooghe, Brianna Laugher, Florian Bruhin and others"
         ),
         entry_points={"console_scripts": ["pytest=pytest:main", "py.test=pytest:main"]},
-        classifiers=classifiers,
         keywords="test unittest",
         # the following should be enabled for release
-        setup_requires=["setuptools-scm"],
+        setup_requires=["setuptools-scm", "setuptools>30.3"],
```
Comment (nicoddemus) [inline setup.py]: This should be `>=` right?

File: setup.py [hunk 2]
```diff
@@ -1,96 +1,25 @@
...
-        if sys.version_info < (3, 6):
-            install_requires.append("pathlib2>=2.2.0")
 
     setup(
-        name="pytest",
-        description="pytest: simple powerful testing with Python",
-        long_description=long_description,
         use_scm_version={"write_to": "src/_pytest/_version.py"},
         url="https://docs.pytest.org/en/latest/",
         project_urls={
```
Comment (nicoddemus) [inline setup.py]: Any other metadata that we can add to `setup.cfg`? Like `license`, `platforms`, `author`, `entry_points`, ...?

Comment (RonnyPfannschmidt) [reply setup.py]: yes

File: setup.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply setup.py]: yikes, good find

**Pull request merged**